### PR TITLE
Update dockerfiles

### PR DESCRIPTION
**Pull request by billingsley-john**
## Proposed changes

PR which re-organises the docker build which occur on PRs. 

On push/PR to develop:
- ukaea/paramak:dev image is built which includes all paramak dependencies + paramak code on develop branch
- ukaea/paramak:dependencies image is built which includes all paramak dependencies
- ukaea/paramak:geometry-dependencies image is built which includes all paramak dependencies (excluding neutronics)

On push/PR to main:
- ukaea/paramak:stable image is built which includes all paramak dependencies + paramak code on main branch
- ukaea/paramak:dependencies image is built which includes all paramak dependencies
- ukaea/paramak:geometry-dependencies image is built which includes all paramak dependencies (excluding neutronics)

I.e. ukaea/paramak:dependencies and geometry-dependencies are built on both PRs to develop and main, so the image will be up-to-date with the most up-to-date of those branches.

All images build CQ2.1 which introduces stability into the paramak; however, this can be changed easily.

## Types of changes

What types of changes does your code introduce to the Paramak?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Pep8 applied
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: .github/workflows/dockerhub-publish-geometry-dependencies.yml [hunk 1]
```diff
@@ -2,11 +2,13 @@
 # This Action will build and upload a Docker image to Dockerhub
 # https://github.com/marketplace/actions/publish-docker
 
-name: dockerhub-publish-dev
+name: dockerhub-publish-geometry-dependencies
 
 on:
   push:
-    branches: develop
+    branches:
```
Comment (shimwell) [inline .github/workflows/dockerhub-publish-geometry-dependencies.yml]:
```suggestion
    branches:
```

File: .github/workflows/dockerhub-publish-geometry-dependencies.yml [hunk 2]
```diff
@@ -2,11 +2,13 @@
...
 # https://github.com/marketplace/actions/publish-docker
 
-name: dockerhub-publish-dev
+name: dockerhub-publish-geometry-dependencies
 
 on:
   push:
-    branches: develop
+    branches: 
+    - develop
```
Comment (shimwell) [inline .github/workflows/dockerhub-publish-geometry-dependencies.yml]:
```suggestion
      - develop
```

File: .github/workflows/dockerhub-publish-geometry-dependencies.yml [hunk 3]
```diff
@@ -2,11 +2,13 @@
...
+    - main
```
Comment (shimwell) [inline .github/workflows/dockerhub-publish-geometry-dependencies.yml]:
```suggestion
      - main
```

**Review by shimwell — APPROVED** [review]
Looks like another nice addition, just a few things code inspector picked up

**Comment by billingsley-john** [comment]
> Looks like another nice addition, just a few things code inspector picked up

Thank you for the suggestions! @Shimwell

**Pull request merged**